.github/workflows/build-docker-image.yml

```diff
@@ -32,33 +32,51 @@ jobs:
       - name: go-test
         run: go test -v ./...
 
-  build:
+  build-image-for-linux:
     runs-on: ubuntu-latest
     needs: test
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-go@v5
-      with:
-        go-version: '1.21'
-    - run: go version
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+      - run: go version
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker info
+        run: docker info
+      - name: Buildx inspect
+        run: docker buildx inspect
+
+      - name: Build image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: linux/Dockerfile
+          platforms: linux/amd64,linux/arm/v7,linux/arm64,linux/ppc64le,linux/s390x,linux/386,linux/arm/v6
+          push: false
+          # Use a 'temp' tag, that won't be pushed for non-release builds
+          tags: testcontainers/ryuk:${{ github.event.release.tag_name || 'temp' }}-linux
 
-    - name: Set up QEMU
-      uses: docker/setup-qemu-action@v3
-    - name: Set up Docker Buildx
-      id: buildx
-      uses: docker/setup-buildx-action@v2
+  build-image-for-windows-2022:
+    runs-on: windows-2022
+    needs: test
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.21'
+      - run: go version
 
-    - name: Docker info
-      run: docker info
-    - name: Buildx inspect
-      run: docker buildx inspect
+      - name: Docker info
+        run: docker info
 
-    - name: Build image
-      uses: docker/build-push-action@v3
-      with:
-        context: .
-        file: ./Dockerfile
-        platforms: linux/amd64,linux/arm/v7,linux/arm64,linux/ppc64le,linux/s390x
-        push: false
-        # Use a 'temp' tag, that won't be pushed, for non-release builds
-        tags: testcontainers/ryuk:${{ github.event.release.tag_name || 'temp' }}
+      - name: Build image
+        # Use a 'temp' tag, that won't be pushed for non-release builds
+        run: |
+          docker build -f windows/Dockerfile --build-arg BASE_IMAGE=mcr.microsoft.com/windows/nanoserver:ltsc2019 -t testcontainers/ryuk:${{ github.event.release.tag_name || 'temp' }}-windows.amd64 .
```

Comment:
Do you think we can build the image for 2022 as well? I know Windows container [version compatibility](https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility) is complex, but supporting both would be awesome, especially considering process isolation and the upcoming support in Windows 11. Otherwise, 2019 is a reasonable choice to support most OS versions.

Reply:
I've added the variant, but didn't check the result, yet.